storage/storage.go

```diff
@@ -1897,16 +1897,57 @@ func applySourceCondsProto(gen int64, conds *Conditions, call *storagepb.Rewrite
 	return nil
 }
 
-// setConditionField sets a field on a *raw.WhateverCall.
+// setGeneration sets Generation on a *raw.WhateverCall.
 // We can't use anonymous interfaces because the return type is
 // different, since the field setters are builders.
-func setConditionField(call reflect.Value, name string, value interface{}) bool {
-	m := call.MethodByName(name)
-	if !m.IsValid() {
-		return false
+// We also make sure to supply a compile-time constant to MethodByName
+// so an not to disable the DCE.
+func setGeneration(cval reflect.Value, value interface{}) bool {
+	m := cval.MethodByName("Generation")
+	if m.IsValid() {
+		m.Call([]reflect.Value{reflect.ValueOf(value)})
 	}
-	m.Call([]reflect.Value{reflect.ValueOf(value)})
-	return true
+	return m.IsValid()
```

Comment:
Can we keep these lines in a single method?

Reply:
Done.